pkg/summary/gvk.go

```diff
@@ -0,0 +1,94 @@
+package summary
+
+import (
+	"encoding/json"
+	"errors"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// conditionTypeStatusJSON is a custom JSON to map a complex object into a standard JSON object. It maps Groups, Versions and Kinds to
+// Conditions, Types, and Status, indicating with a flag if a certain condition with specific status represents an error or not.
+// It is expected to be something like:
+//
+//	{
+//			"gvk": 			"helm.cattle.io/v1, Kind=HelmChart",
+//			"conditionMappings": [
+//				{
+//					"type": "JobCreated"	// This means JobCreated is mostly informational and True or False doesn't mean error
+//				},
+//				{
+//					"type": "Failed",	// This means Failed is considered error if it's status is True
+//					"status": ["True"]
+//				},
+//			}
+//	}
+type conditionTypeStatusJSON struct {
+	GVK               string                     `json:"gvk"`
+	ConditionMappings []conditionStatusErrorJSON `json:"conditionMappings"`
+}
+
+type conditionStatusErrorJSON struct {
+	Type   string                   `json:"type"`
+	Status []metav1.ConditionStatus `json:"status,omitempty"`
+}
+
+type ConditionTypeStatusErrorMapping map[schema.GroupVersionKind]map[string]sets.Set[metav1.ConditionStatus]
+
+func (m ConditionTypeStatusErrorMapping) MarshalJSON() ([]byte, error) {
+	output := []conditionTypeStatusJSON{}
```

Comment:
There must be a better name for the variable you're using to gather the json than `output`. Maybe `statuses`?  `output` implies something is getting emitted somewhere.

Reply:
> output implies something is getting emitted somewhere.

That is exactly what's happening. `conditionTypeStatusJSON` is the intermediate type used for translation between the internal maps and the JSON structure. That structure is what is actually marshalled and output.